crabs/detection_tracking/models.py

```diff
@@ -156,17 +156,26 @@ def on_validation_epoch_end(self) -> None:
         """
         Hook called after each validation epoch to compute metrics and logging.
         """
-        self.validation_step_outputs = self.compute_precision_recall_epoch(
+        (
+            self.validation_step_outputs,
+            val_precision,
+            val_recall,
+        ) = self.compute_precision_recall_epoch(
             self.validation_step_outputs, "val"
         )
 
+        self.log("val_precision", val_precision)
```

Comment:
Question: should we use `self.logger.log_metrics` instead?

Reply:
logger.log_metrics not working for trainer.callback_metrics which what this log for, to be used during optuna trial.

Reply:
aaah ok. Maybe we make a if-optuna case to make it clear?